src/tp_engine.cpp

```diff
@@ -159,23 +443,164 @@ void TPEngine::GrowSpareEdgeData(const size_t new_edge_spare_count) {
   }
 }
 
-void TPEngine::InitializeBranchLengthsByTakingFirst(
+// ** Tree Collection
+
+void TPEngine::FunctionOverRootedTreeCollection(
```

Comment:
This is copied from somewhere else, right? Why do we need it here?

Reply:
I felt hesistant to make a dependency of `TPEngine` on `GPEngine` (where this function lives currently).  Should I do that?  Or maybe I should just move the function somewhere, like to `tree_collection.hpp`?

Reply:
I totally agree about not wanting to set up that dependency, thanks!

I like the idea of moving this as a free function. If it works in tree_collection that's good, if not what about `sbn_maps.cpp`?

In general, lets think twice / discuss before doing a copy-paste. I totally understand that it keeps things moving in the short term, which is great, but just mark with a TODO to revise before the PR stage.

Reply:
Okay.  Moved `FunctionOverRootedTreeCollection` to `RootedSBNMap` namespace.

Reply:
I think this function was written for the top pruning proof of concept I did in the watching-mb and isn't used elsewhere. So I agree it makes sense to pull it away from GPEngine.